model/load_balancer.rb

```diff
@@ -117,11 +117,11 @@ def dns_zone
   def need_certificates?
     return true if certs_dataset.empty?
 
-    certs_dataset.where { created_at > Time.now - 60 * 60 * 24 * 30 * 2 }.empty?
+    certs_dataset.where { created_at > Time.now - 60 * 60 * 24 * 30 * 2 }.exclude(cert: nil).empty?
```

Comment:
is it guaranteed that other required fields are also set (like key) when the cert is set?

Reply:
Yes, cert is the very last field we get following the acme challenge.